src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -64,7 +79,8 @@ public boolean equals(Object other) {
         Appointment otherAppointment = (Appointment) other;
         return appointmentDatetime.equals(otherAppointment.appointmentDatetime)
                 && seller.equals(otherAppointment.seller)
-                && buyer.equals(otherAppointment.buyer);
+                && ((buyer != null && buyer.equals(otherAppointment.buyer))
+                    || buyer == otherAppointment.buyer);
```

Comment:
While reviewing the code, I felt the readability was off.
Would it be possible to simply the above method to
```suggestion
                && Objects.equals(buyer, otherAppointment.buyer);
```
This would also handle all null cases while improving readability.